### PR TITLE
Add link check

### PR DESCRIPTION
**Pull request by rogermparent**
This PR adds our `repo-link-check` package in both GitHub Actions and local npm scripts, with the same implementation that dvc.org currently has.

### CODE REVIEW

**Comment by rogermparent** [comment]
Looks like it's acting a bit weird next to the lint CI
image

File: config/link-check/config.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+rootURL: https://dvc.org
```
Comment (julieg18) [inline config/link-check/config.yml:1]: Does this rootURL need to be changed to mlem.ai?

File: .github/workflows/link-check-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          rootURL: '${{ github.event.deployment.payload.web_url }}'
+          output: checksAction
+      - uses: LouisBrunner/checks-action@v1.0.0
+        if: ${{ success() }}
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          check_id: ${{ steps.build_check.outputs.check_id }}
+          status: completed
+          conclusion: ${{ steps.check.outputs.conclusion }}
+          output: ${{ steps.check.outputs.output }}
```
Comment (julieg18) [inline .github/workflows/link-check-deploy.yml:39]: Is there a way to name the report check? This is pretty unclear that's it's checking links 🤔 

image

File: .github/workflows/link-check-deploy.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+name: Check new links against deployment
```
Comment (julieg18) [inline .github/workflows/link-check-deploy.yml:1]: Do we want a daily link check as well? 🤔

**Review by julieg18 — COMMENTED** [review]
We should probably open up a test pr with a failing link to make sure things are working correctly, either before or after this is merged :)

File: .github/workflows/link-check-deploy.yml (hunk 2 above)
Comment (rogermparent) [reply .github/workflows/link-check-deploy.yml:39]: I'm not sure what's going on, seems to be a GH Actions bug where this action's report is taking the name of another action. Maybe doing a CLI implementation will solve the issue by making there less moving parts in check creation.

File: config/link-check/config.yml (hunk 1 above)
Comment (rogermparent) [reply config/link-check/config.yml]: Only for the daily check, but good point nonetheless!

**Comment by rogermparent** [comment]
I've added some broken links to a doc to test out the integration, and while those are in there I'll mark the PR draft so we can't accidentally merge them in. After we ensure it works, I'll remove them and put the PR back to open.

**Comment by rogermparent** [comment]
Looks like it works!

image

image

File: .github/workflows/link-check-deploy.yml (hunk 3 above)
Comment (rogermparent) [reply .github/workflows/link-check-deploy.yml:1]: Good point! Added that in.

**Comment by rogermparent** [comment]
CLI-based check works as well, and doesn't have the name issue:

image

image

**Review by julieg18 — APPROVED** [review]
Looks good!

**Pull request merged**